runtime-modules/content/src/lib.rs

```diff
@@ -1259,8 +1307,72 @@ decl_module! {
         ) {
             Self::not_implemented()?;
         }
+
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_commitment(
+            origin,
+            new_commitment: <T as frame_system::Trait>::Hash,
+        ) {
+        ensure_root(origin)?;
+            let old_commitment = <Commitment<T>>::get();
+            if old_commitment == new_commitment { return Ok(()) }
+            <Commitment<T>>::put(new_commitment);
+            Self::deposit_event(RawEvent::CommitmentUpdated(new_commitment));
+        }
+
+    #[weight = 10_000_000]
+    pub fn update_channel_reward(
```

Comment:
Check the structure of other extrinsics, there should be a clear distinction between checking and mutation, and we introduce a `mutation safe` marker at the top lexical scope of each extrinsic to show this.

Reply:
1) Lets rename this to `claim_channel_reward`.
2) The origin for this has to be the owner of the channel, and since we should only allow channels owned by members, vs. those owned by curators, the owner has to be a member. So signature needs a member id. Other extrinsics where the caller can be either a member or a curator, use `actor: ContentActor`, but this is not appropriate here.